### ISSUE TITLE
Unable to get articles in page object 

### ISSUE THREAD

**Issue by Yilun-Sun** [honkit/honkit#241]
Hi, all. When I tried to generate the articles which are in the child hierarchy of the current page, I found there is no such `articles` attribute in the `page` object. But `page.next` and `page.previous` has. So I turned to the inelegant solution which is to get the `articles` by something similar to `page.next.previous.articles`.

I wonder if I could add the `articles` attribute to the `page` in `encodePage.ts` and create a **PR** for that. Like:
```typescript
function encodePage(page: Page, summary: Summary): EncodedPage {
    ...

    if (article) {
        ...

        const prevArticle = summary.getPrevArticle(article);
        if (prevArticle) {
            result.previous = encodeSummaryArticle(prevArticle);
        }

        // add articles to result
        const articles = article
            .getArticles()
            .map((article) => encodeSummaryArticle(article))
            .toJS();
        if (articles.length > 0) {
            result.articles = articles;
        }
    }

    ...

    return result;
}
```

And what is the reason of the `articles` property is in the `page.next` but not in the `page`? When will `page.next.articles` be used?

Thanks!

**Comment by azu** [comment]
This consistency is resonable.

**Comment by azu** [comment]
📝 I've noticed `encodeSummaryArticle` is recurcive. I feel like there is waste.
We may need to memorize the `encodeSummaryArticle` function if performance adverse effect.
https://github.com/honkit/honkit/blob/be282ebe68ae63a2b7ca333b33147c0914dce41b/packages/honkit/src/json/encodeSummaryArticle.ts#L10-L17

**Comment by Yilun-Sun** [comment]
> 📝 I've noticed `encodeSummaryArticle` is recurcive. I feel like there is waste. ( Probably,`encodeSummaryArticle` encode same `article` many times) We may need to memorize the `encodeSummaryArticle` function if performance adverse effect.
> 
> https://github.com/honkit/honkit/blob/be282ebe68ae63a2b7ca333b33147c0914dce41b/packages/honkit/src/json/encodeSummaryArticle.ts#L10-L17

Thanks for your reply! Yes, by memorizing the encoded summary, the performance could be easily improved.
I tried to measure `encodeSummaryArticle` calls for my **105-pages** article:

| condition | call times | time to serve |
|  ----  | ---- | ----  |
| no changes | 112773 | 49.6s |
| use memorizeOne | 112767 | 52.55s |
| HashMap | 105 | 21.6s |

The HashMap works like a charm. I will create a PR for this fix, please check once you are available, thanks a lot! Also, I include some doc updates.

**Comment by Yilun-Sun** [comment]
Hi @azu . I just created two **PR**s, please review them once you have time, thank you so much, appreciate that! =D

**Comment by azu** [comment]
It was released in v3.7.1 #250 

Thanks!
